File: Geometry/Q1. Projection Area of 3D Shapes/Q1_ProjectionAreaof3DShapes.java
class Solution {
    public int projectionArea(int[][] grid) {
        int n = grid.length;              // get the size of the grid
        
        int xy = 0;                       // area from top view
        int yz = 0;                       // area from front view
        int zx = 0;                       // area from side view
        
        for (int i = 0; i < n; i++) {     // loop each row
            int rowMax = 0;               // max value in this row (for zx)
            int colMax = 0;               // max value in this column (for yz)
            
            for (int j = 0; j < n; j++) { // loop each column
               
                if (grid[i][j] > 0)       // if cell has cube(s)
                    xy++;                 // count it for top view
                
                rowMax = Math.max(rowMax, grid[i][j]); // update row max
                
                colMax = Math.max(colMax, grid[j][i]); // update column max
            }
            
            zx += rowMax;                 // add row max to side view area
            yz += colMax;                 // add column max to front view area
        }
        
        return xy + yz + zx;              // total projection area
    }
}

File: Geometry/Q1. Projection Area of 3D Shapes/Q1_ProjectionAreaof3DShapes.py
class Solution(object):
    def projectionArea(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        
        n = len(grid)                      # size of grid
        
        xy = 0                              # top view area
        yz = 0                              # front view area
        zx = 0                              # side view area
        
        for i in range(n):                 # loop each row
            row_max = 0                    # max in this row (for side view)
            col_max = 0                    # max in this column (for front view)
            
            for j in range(n):             # loop each column
                if grid[i][j] > 0:         # if cell has cube(s)
                    xy += 1                # count for top view
                
                row_max = max(row_max, grid[i][j])   # update row max
                col_max = max(col_max, grid[j][i])   # update column max
            
            zx += row_max                  # add row max to side view
            yz += col_max                  # add column max to front view
        
        return xy + yz + zx                # return total area
